app/uk/gov/hmrc/customs/declarations/stub/services/DeclarationStatusResponseBuilder.scala

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.customs.declarations.stub.services
+
+import uk.gov.hmrc.customs.declarations.stub.models.declarationstatus.DeclarationStatusResponse
+import uk.gov.hmrc.customs.declarations.stub.models.declarationstatus.DeclarationStatusResponse._
+
+class DeclarationStatusResponseBuilder {
+
+  def buildDeclarationStatus(eori: String, mrn: String): DeclarationStatusResponse = mrn match {
+    case mrn if mrn.endsWith("9999") => NotFoundResponse
+    case _                           => SuccessfulResponse(successfulResponseBody(eori, mrn))
+  }
```

Comment:
I know that the team likes pattern matching :smiley: but this looks to me a bit over engineered. What's wrong with ?
```
if (mrn.endsWith("9999")) NotFoundResponse
else SuccessfulResponse(successfulResponseBody(eori, mrn))
```

Reply:
There is nothing wrong with it. Thought about the same but eventually sticked to pattern matching.